NewForce-at-Mountwest/bangazon-api-gnocci
# Josh branch

# Description

Created Git Method
Created Get single to get Order by id.
Created Post method to create a new Order.
Created Put method to update a Order.
Delete method to delete a Order.

Created Git Method
Created Get single to get ProductType by id.
Created Post method to create a new ProductType.
Created Put method to update a ProductType.
Hard/Soft Delete method. Hard delete for testing and soft delete to archive ProductType


## Related Ticket(s)
(Orders)
4
(ProductType)
5

## Steps to Test
Steps on how to run code:
Open GitBash
Clone the repo
cd bangazon-api-gnocci
cd bangazonApi
start bangazonApi.sln
--once the code is opened in Visual Studio 2019
F5 to run the code
https://localhost:5001/api/order <- to get all Order
https://localhost:5001/api/order/1 <- to get single Order
https://localhost:5001/api/order?include=customers <- to include customers
https://localhost:5001/api/order?include=products <- to include products
https://localhost:5001/api/order?completed=true <- to only see completed orders
https://localhost:5001/api/order?completed=false <- to only see not completed orders

https://localhost:5001/api/ProductType <- to get all ProductTypes
https://localhost:5001/api/ProductType/1 <- To get single ProductTypes




# Checklist:

Sorry I'm stupid and did orders and producttypes on the same branch then pushed up producttypes thinking I could make a new PR but It just updated my current PR code ):

-- SQL CODE FOR ORDERPRODUCT --

INSERT INTO OrderProduct(OrderId, ProductId) VALUES (1, 2);
INSERT INTO OrderProduct(OrderId, ProductId) VALUES (1, 2);
INSERT INTO OrderProduct(OrderId, ProductId) VALUES (2, 3);
INSERT INTO OrderProduct(OrderId, ProductId) VALUES (3, 4);
INSERT INTO OrderProduct(OrderId, ProductId) VALUES (4, 5);
INSERT INTO OrderProduct(OrderId, ProductId) VALUES (5, 6);
